exercise_3/main.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include "ShowCommand.hpp"
+
+const std::string EXIT = "EXIT";
+const std::string HELP = "HELP";
+
+namespace {
+    auto show_help = [] {
+        std::cout << "Available commands:" << std::endl;
+        std::cout << "\tEXIT: Type to quit." << std::endl;
+        std::cout << "\tHELP: Type to print this message." << std::endl;
+        std::cout << std::endl;
+        std::cout << "\tSHOW [table_name]: If table_name is present - print table or print all tables." << std::endl;
+    };
+}
+
+static void select_command(const std::string &command, std::istringstream &istringstream, RecordsHandler &recordsHandler) {
+    if (ShowCommand::can_execute(command)) {
```

Comment:
That's not really scalable.

Consider having at least 100 different commands.
That would require you to provide 100 public static methods, 100 public static method's implementation and 100 if-statements.

Why not simply provide any form of an abstract method pattern implementation? That should ease the code structure.

Reply:
What is the best way to select which command should be used for execution?

Reply:
Kind of: https://godbolt.org/z/U2Yi9w